lightning/src/chain/channelmonitor.rs

```diff
@@ -1530,9 +1635,50 @@ impl<Signer: Sign> ChannelMonitor<Signer> {
 						// confirmation with the same height or have never met our dust amount.
 					}
 				}
+				if Some(txid) == us.current_counterparty_commitment_txid || Some(txid) == us.prev_counterparty_commitment_txid {
+					walk_htlcs!(false, false, counterparty_tx_htlcs.iter().map(|(a, _)| a));
+				} else {
+					walk_htlcs!(false, true, counterparty_tx_htlcs.iter().map(|(a, _)| a));
+					// The counterparty broadcasted a revoked state!
+					// Look for a StaticOutput spend first, as it should be spending the full set
+					// of commitment transaction outputs, if we see one, assume that it did and
+					// just return that.
```

Comment:
This comment may need amending since we do return another RevokedOutput balance under some circumstances